test/e2e/far_e2e_test.go

```diff
@@ -86,48 +85,12 @@ var _ = Describe("FAR E2e", func() {
 		testNodeParam, err = buildNodeParameters(clusterPlatform.Status.PlatformStatus.Type)
 		Expect(err).ToNot(HaveOccurred(), "can't get node information")
 	})
-	Context("stress cluster with ResourceDeletion remediation strategy", func() {
-		var availableWorkerNodes *corev1.NodeList
-		BeforeEach(func() {
-			if availableWorkerNodes == nil {
-				availableWorkerNodes = getAvailableWorkerNodes()
-			}
-			selectedNode = pickRemediatedNode(availableWorkerNodes)
-			nodeName = selectedNode.Name
-			printNodeDetails(selectedNode, nodeIdentifierPrefix, testNodeParam)
 
-			// save the node's boot time prior to the fence agent call
-			nodeBootTimeBefore, err = e2eUtils.GetBootTime(clientSet, nodeName, testNsName, log)
-			Expect(err).ToNot(HaveOccurred(), "failed to get boot time of the node")
-
-			// create tested pod which will be deleted by the far CR
-			pod = createTestedPod(nodeName, testContainerName)
-			DeferCleanup(cleanupTestedResources, pod)
-
-			// set the node as "unhealthy" by disabling kubelet
-			makeNodeUnready(selectedNode)
-
-			startTime = time.Now()
-			remediationStrategy = v1alpha1.ResourceDeletionRemediationStrategy
-			far := createFAR(nodeName, fenceAgent, testShareParam, testNodeParam, remediationStrategy)
-			DeferCleanup(deleteFAR, far)
-		})
-		When("running FAR to reboot two nodes", func() {
-			It("should successfully remediate the first node", func() {
-				checkRemediation(nodeName, nodeBootTimeBefore, pod, remediationStrategy)
-				remediationTimes = append(remediationTimes, time.Since(startTime))
-			})
-			It("should successfully remediate the second node", func() {
-				checkRemediation(nodeName, nodeBootTimeBefore, pod, remediationStrategy)
-				remediationTimes = append(remediationTimes, time.Since(startTime))
-			})
-		})
-	})
-	Context("stress cluster with OutOfServiceTaint remediation strategy", func() {
+	runFARTests := func(remediationStrategy v1alpha1.RemediationStrategyType, skipCondition func() bool) {
```

Comment:
Maybe I'm missing something, but can we just use a normal function here?

```go
func runFARTest(...) bool {
...
}

Reply:
I've seen and used this pattern especially in tests more often. Advantages IMHO:

- the function is "physically" located nearer to the code where it's used, which improves readability IMHO
- the function has direct access to vars of the context, so less args are needed
- often these functions are very special for the tests of a specific context. Making them a regular `func` would make them available for all tests though, which isn't needed

Reply:
Fair enough, could we make it stand up a bit more in the context then? Even some space around might be enough

Reply:
@slintes Thank you for the explanation. This is everything I thought.
In the current code, the test routine I wrote is duplicated and it will be troubled for maintenance in the future.
To remove the duplicated code, it needs access to vars of the context. This is the reason why I used this pattern.
@clobrano 
Thank you for the comment. I added the following comment. Is that fine for you?
```
+       // runFARTests is a utility function to run FAR tests.
+       // It accepts a remediation strategy and a condition to determine if the tests should be skipped.
+       runFARTests := func(remediationStrategy v1alpha1.RemediationStrategyType, skipCondition func() bool) {
```

Reply:
Yes, thank you